client/utils/currency/index.js

```diff
@@ -52,12 +52,23 @@ export const formatCurrency = ( amount, currencyCode ) => {
 
 	const currency = getCurrency( currencyCode );
 	if ( null === currency ) {
-		// Fallback for unsupported currencies: currency code and amount
-		return sprintf(
-			isZeroDecimal ? '%s %i' : '%s %.2f',
-			currencyCode.toUpperCase(),
-			amount
-		);
+		return composeFallbackCurrency( amount, currencyCode, isZeroDecimal );
+	}
+
+	try {
+		return 'function' === typeof currency.formatAmount
+			? currency.formatAmount( amount )
+			: currency.formatCurrency( amount );
+	} catch ( err ) {
+		return composeFallbackCurrency( amount, currencyCode, isZeroDecimal );
 	}
-	return currency.formatCurrency( amount );
 };
+
+function composeFallbackCurrency( amount, currencyCode = '', isZeroDecimal ) {
```

Comment:
We don't need default value for `currencyCode` here, it's provided in all given invocations.

Reply:
Good point, fixed in https://github.com/Automattic/woocommerce-payments/pull/1236/commits/4888871975ce299e66c1c9eb1085c2cba7b89fed